#166978014 - Testing All endpoints on Heroku

UhiriweAudace opened the pull request:
#### What does this PR do?
- Testing all endpoints for ensuring that are working as expected
- Documenting all discovered errors
#### Description of Task to be completed?
#### How should this be manually tested?
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[166978014](https://www.pivotaltracker.com/n/projects/2350798/stories/166978014)
#### Screenshots (if appropriate)

#### Questions:

Commit:
Resolve conflict

Commit:
Resolve conflict

UhiriweAudace committed:
[chore #166978014] Testing All endpoints on heroku

UhiriweAudace committed:
[chore #166978014] Testing All endpoints on heroku

Quantum-35 commented on src/api/controllers/articlesController.js:
```diff
@@ -92,7 +92,7 @@ class articlesController {
 
     if (!allArticle[0]) {
-      return res.status(404).send({ error: 'Whoops! No Articles found!' });
+      return res.status(404).send({ message: 'Whoops! No Articles found!' });
```
I prefer `error` as the key to displaying errors.

UhiriweAudace replied:
@Quantum-35      ==> the response returns an empty array. can i change the status code from 404 to 200 with a message instead of error? could I have your attention on this, please?

Commit:
feat(upload): upload image

- user can upload image
- frontend integration tweaks
- fix following and unfollowing logic
- update shares

Finishes [#167533974]

Commit:
feat(upload): upload image

- user can upload image
- frontend integration tweaks
- fix following and unfollowing logic

Finishes [#167533974]

Commit:
feat(upload): upload image

- user can upload image
- frontend integration tweaks

Finishes [#167533974]

Commit:
feat(upload): upload image

- user can upload image
- frontend integration tweaks
- fix following and unfollowing logic

Finishes [#167533974]

Commit:
feat(upload): upload image

- user can upload image
- frontend integration tweaks
- fix following and unfollowing logic

Finishes [#167533974]

rukundoeric commented:
Kindly remove those violation and take care about those test which are failing

rukundoeric reviewed (changes requested):
Kindly remove those violation and take care about those test which are failing

Commit:
Added a chat handler connected to the server port

Commit:
Added a chat handler connected to the server port

Commit:
Added a chat handler connected to the server port

Commit:
Ceiling the percentage of ratings

eliemugenzi committed:
Merge pull request #84 from andela/bg-rating-fix

Bg rating fix

Commit:
Fix username uniqueness issue

Commit:
remove username middleware
